### ISSUE TITLE
proxy cache:Error response from daemon: received unexpected HTTP status: 500 Internal Server Error

### ISSUE THREAD

**Issue by love-miao** [goharbor/harbor#14588]
I use helm(1.6.1) install two harbor(2.2.1), one is source,one is proxy cache.

source is https://registry-1.docker.io proxy cache is work ok

**but local harbor from proxy cache docker pull error:**
`proxy cache:Error response from daemon: received unexpected HTTP status: 500 Internal Server Error`

**core logs:**
```
2021-04-07T09:43:45Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"http status code: 400, body: "}]}
/harbor/src/pkg/registry/client.go:546, github.com/goharbor/harbor/src/pkg/registry.(*client).do
/harbor/src/pkg/registry/client.go:253, github.com/goharbor/harbor/src/pkg/registry.(*client).ManifestExist
/harbor/src/controller/proxy/remote.go:90, github.com/goharbor/harbor/src/controller/proxy.(*remoteHelper).ManifestExist
/harbor/src/controller/proxy/controller.go:151, github.com/goharbor/harbor/src/controller/proxy.(*controller).UseLocalManifest
/harbor/src/server/middleware/repoproxy/proxy.go:119, github.com/goharbor/harbor/src/server/middleware/repoproxy.handleManifest
/harbor/src/server/middleware/repoproxy/proxy.go:99, github.com/goharbor/harbor/src/server/middleware/repoproxy.ManifestMiddleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/metric/metric.go:84, github.com/goharbor/harbor/src/server/middleware/metric.InjectOpIDMiddleware.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/v2auth/auth.go:167, github.com/goharbor/harbor/src/server/middleware/v2auth.Middleware.func2.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/router/router.go:90, github.com/goharbor/harbor/src/server/router.(*Route).Handler.func1
/harbor/src/vendor/github.com/astaxie/beego/router.go:769, github.com/astaxie/beego.(*ControllerRegister).ServeHTTP
/harbor/src/server/middleware/middleware.go:52, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/security/security.go:76, github.com/goharbor/harbor/src/server/middleware/security.UnauthorizedMiddleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/security/security.go:63, github.com/goharbor/harbor/src/server/middleware/security.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/artifactinfo/artifact_info.go:90, github.com/goharbor/harbor/src/server/middleware/artifactinfo.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/transaction/transaction.go:62, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1.1
/harbor/src/lib/orm/orm.go:76, github.com/goharbor/harbor/src/lib/orm.WithTransaction.func1
/harbor/src/server/middleware/transaction/transaction.go:71, github.com/goharbor/harbor/src/server/middleware/transaction.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/notification/notification.go:31, github.com/goharbor/harbor/src/server/middleware/notification.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/orm/orm.go:53, github.com/goharbor/harbor/src/server/middleware/orm.MiddlewareWithConfig.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/middleware.go:52, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/session/session.go:34, github.com/goharbor/harbor/src/server/middleware/session.Middleware.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/log/log.go:33, github.com/goharbor/harbor/src/server/middleware/log.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/requestid/requestid.go:37, github.com/goharbor/harbor/src/server/middleware/requestid.Middleware.func1
/harbor/src/server/middleware/middleware.go:57, github.com/goharbor/harbor/src/server/middleware.New.func1.1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/harbor/src/server/middleware/metric/metric.go:67, github.com/goharbor/harbor/src/server/middleware/metric.transparentHandler.func1
/usr/local/go/src/net/http/server.go:2042, net/http.HandlerFunc.ServeHTTP
/usr/local/go/src/net/http/server.go:2843, net/http.serverHandler.ServeHTTP
/usr/local/go/src/net/http/server.go:1925, net/http.(*conn).serve
```

**Comment by stonezdj** [comment]
There are two Harbors installed, one is the proxy_cache, and another is the source registry, it seems that the proxy cache sends a HEAD request to the source registry and got a 400 error (bad request), could you please collect the core log in the source registry which contains the 400 error?

**Comment by love-miao** [comment]
yes include 400

**Comment by stonezdj** [comment]
Fixed in #14949
